app/models/course_template.rb

```diff
@@ -0,0 +1,27 @@
+# Course stub to be copied by teachers for their own organisations
+
+require 'net/http'
+
+class CourseTemplate < ActiveRecord::Base
+  validates :name,
+            presence: true,
+            uniqueness: true,
+            length: { within: 1..40 },
+            format: {
+              without: / /,
+              message: 'should not contain white spaces'
+            }
+  validates :source_url, presence: true
+  validates :title,
+            presence: true,
+            length: { within: 4..40 }
+  validates :description, length: { maximum: 512 }
+  validate :valid_source_url?
+
+  def valid_source_url?
+    url = URI(source_url)
+    Net::HTTP.get(url)
```

Comment:
Not sure if making a full request to the uri is wanted (change for misuse - what if the site returns a huge page?)